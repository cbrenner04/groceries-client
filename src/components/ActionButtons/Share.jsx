import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

const Share = ({ to, ...rest }) => (
  <Button href={to} variant="link" className="p-0 mr-3" {...rest}>
    <i className="fa fa-users fa-2x text-primary" />
  </Button>
);

Share.propTypes = {
  to: PropTypes.string.isRequired,
};

export default Share;
