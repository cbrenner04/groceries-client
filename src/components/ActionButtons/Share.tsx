import React, { type CSSProperties } from 'react';
import { Link } from 'react-router';

interface IShareProps {
  to: string;
  testID: string;
  disabled: boolean;
  style?: CSSProperties;
  classes?: string;
}

const Share: React.FC<IShareProps> = (props): React.JSX.Element => {
  const providedStyles = props.style ?? {};
  const updatedStyles: CSSProperties = props.disabled ? { ...providedStyles, pointerEvents: 'none' } : providedStyles;
  return (
    <Link
      className={`p-0 me-3 ${props.classes ?? ''}`}
      to={props.to}
      data-test-id={props.testID}
      style={updatedStyles}
      aria-disabled={props.disabled}
    >
      <i className="fa fa-users fa-2x text-primary" />
    </Link>
  );
};

export default Share;
