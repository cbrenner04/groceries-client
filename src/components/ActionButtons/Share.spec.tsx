import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Share from './Share';

async function setup(): Promise<{ shareButton: HTMLElement }> {
  const props = {
    to: '/fake/route',
    testID: 'foo',
    disabled: false,
    style: {},
  };
  const { findByRole } = render(
    <MemoryRouter>
      <Share {...props} />
    </MemoryRouter>,
  );
  const shareButton = await findByRole('link');

  return { shareButton };
}

describe('Share', () => {
  it('renders a link', async () => {
    const { shareButton } = await setup();

    expect(shareButton).toMatchSnapshot();
    expect(shareButton).toHaveAttribute('href', '/fake/route');
  });
});
