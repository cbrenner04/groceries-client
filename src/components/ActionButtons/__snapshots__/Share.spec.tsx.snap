// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share renders a link 1`] = `
<a
  aria-disabled="false"
  class="p-0 me-3 "
  data-test-id="foo"
  href="/fake/route"
>
  <i
    class="fa fa-users fa-2x text-primary"
  />
</a>
`;
